swarm/network/stream/delivery.go

```diff
@@ -208,14 +213,13 @@ R:
 		}
 		chunk.SData = req.SData
 		d.db.Put(chunk)
-		chunk.WaitToStore()
-		err = chunk.GetErrored()
-		if err != nil {
+
+		go func(req *ChunkDeliveryMsg) {
+			err := chunk.WaitToStore()
 			if err == storage.ErrChunkInvalid {
 				req.peer.Drop(err)
 			}
```

Comment:
Maybe I miss something, but we don't we have to close chun.ReqC here?

Reply:
ReqC closing is actually moved to LocalStore.Put.